lang/semgrep-grammars/src/semgrep-julia/grammar.js

```diff
@@ -20,6 +20,28 @@ module.exports = grammar(base_grammar, {
   rules: {
     semgrep_ellipsis: $ => '...',
 
+    // Metavariables
+    identifier: ($, previous) => {
+      return token(
+        choice(
+          previous,
+          // We allow an identifier to be the same as an identifier, but prefixed
+          // with a dollar sign.
+          // We could use a more specific regular expression, which would match
+          // metavariables more faithfully, but either:
+          // 1) this would result in cases where we tokenize as an identifier or
+          // an interpolation expression where we don't want to, depending on how
+          // we set up the precedence.
+          // 2) this causes tokenization issues with interpolation expression that
```

Comment:
This should not be the case, we could use a regexp that ensure we stop at a word boundary (\b in Perl regexp).
I'm sure there's a way in tree-sitter to accept $P but not $Pack.